Chronological/Filter.cs

```diff
@@ -185,14 +186,13 @@ private static string MemberExpressionToString(MemberExpression memberExpression
                 return ExpressionToString(Expression.Constant(result));
             }
 
-            if (memberExpression.Expression.NodeType == ExpressionType.Parameter)
+            if (GetInnermostExpressionType(memberExpression) == ExpressionType.Parameter)
             {
                 var eventFieldMemberExpression = new EventFieldMemberExpression(memberExpression);
-
                 return eventFieldMemberExpression.EscapedEventFieldName;
             }
 
-            if (memberExpression.Expression.NodeType == ExpressionType.Constant)
+            if (GetInnermostExpressionType(memberExpression) == ExpressionType.Constant)
```

Comment:
We really only need to do the recursive bit once and save as a variable, it's probably not that costly but given the other comments we are going to need a third check for `innermost == ExpressionType.MemberAccess` it seems like a sensible optimisation.

Reply:
No problem to save the result as "innermost" variable :)

But for the third check did you mean something like this?
```csharp
if (innermost == ExpressionType.Constant || innermost == ExpressionType.MemberAccess)
```
As based on my first commit, or do you have another idea?

Reply:
Yes exactly, I think it is safe to evaluate the MemberAccess if we are certain we have navigated to the innermost expression. I quickly tested this with a "real" type instead of an anonymous one and this seemed to work. There may be edge cases that I haven't considered but this is going to be better than it was and it will probably throw NotImplemented if there are missing scenarios.

Reply:
Ok, I will apply the fix and commit it in the morning (UTC+1) 😃 .